verilog/analysis/checkers/banned_declared_name_patterns_rule_test.cc

```diff
@@ -0,0 +1,65 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "verilog/analysis/checkers/banned_declared_name_patterns_rule.h"
+
+#include <initializer_list>
+
+#include "gtest/gtest.h"
+#include "common/analysis/linter_test_utils.h"
+#include "common/analysis/syntax_tree_linter_test_utils.h"
+#include "common/text/symbol.h"
+#include "verilog/CST/verilog_nonterminals.h"
+#include "verilog/analysis/verilog_analyzer.h"
+
+namespace verilog {
+namespace analysis {
+namespace {
+
+using verible::LintTestCase;
+using verible::RunConfiguredLintTestCases;
+using verible::RunLintTestCases;
+
+// Tests that BannedDeclaredNamePatternsRuleTest correctly accepts valid identifiers.
+TEST(BannedDeclaredNamePatternsRuleTest, AcceptTests) {
+  const std::initializer_list<LintTestCase> kTestCases = {
+      {""},
+      {"module foo; endmodule"},
+      {"package p; endpackage"},
+
+  };
+  RunLintTestCases<VerilogAnalyzer, BannedDeclaredNamePatternsRule>(kTestCases);
+}
+
+// Tests that BannedDeclaredNamePatternsRuleTest correctly rejects invalid patterns.
+TEST(BannedDeclaredNamePatternsRuleTest, RejectTests) {
+  const std::initializer_list<LintTestCase> kTestCases = {
+      {"module legal;\n"
+       "  module ILLEGALNAME;\n"
+       "  endmodule\n"
+       "endmodule"},
+      {"module legal;\n"
+       "  module illegalname;\n"
+       "  endmodule\n"
+       "endmodule"},
```

Comment:
Suggest also including simpler versions of test cases like "module legal;endmodule", and "module illegalname;endmodule".  Also when you do expect a finding to point to a piece of text, it should be annotated like `{SomeTag, "ILLEGALNAME"}`.  See other rule tests for examples.  It works similarly in concept to the syntax tree match test cases you've worked with.

Reply:
David, please pardon my many questions. I have pushed the simpler and testcases with regions of interest but it is still giving me some errors.. do you think there is something wrong with my implementation? I tested with unit base case as well, but failing one rule and all rule still

Reply:
I'll take a look.  I can reproduce the failures with the current PR.  Something isn't working as expected.

Reply:
I found the problem:

When I said to use `Visit(const SyntaxTreeNode&)`, I meant HandleNode(const SyntaxTreeNode&, const verible::SyntaxTreeContext&)`.  Sorry for the confusion.

Update that function signature in the .h and .cc file.

Reply:
Thanks alot David. so if i understand it correctly, syntaxtreelint rule only generates virtual functions for leaf, node and symbol, so we cannot make new signatures

Reply:
You can make other helper functions in the lint rule classes, but the SyntaxTreeRuleLinter driver expects to call these virtual functions directly.  (There are other kinds of lint rule base classes too, e.g. purely token based, line-based.)